src/psyclone/tests/psyir/nodes/literal_test.py

```diff
@@ -135,6 +135,20 @@ def test_literal_init_invalid_2(value):
             f"found '{value}'." in str(err.value))
 
 
+@pytest.mark.parametrize("value", ["02", "++2", "12.34", "*2"])
```

Comment:
Done - I think i came across fullmatch solely due to trying to work out how to implement this check in the Particle IR as otherwise it ended up accepting anything containing a number :(